packages/modules/src/firebase/functions-checkout/get-custumer-auth.ts

```diff
@@ -0,0 +1,53 @@
+import type { Customers } from '@cloudcommerce/types';
+import api from '@cloudcommerce/api';
+import { logger } from 'firebase-functions';
+// eslint-disable-next-line import/no-unresolved
+import { getFirestore } from 'firebase-admin/firestore';
+import { generateAccessToken, findCustomerByEmail } from '@cloudcommerce/passport';
+
+type CustomerCheckout = {
+  _id?: string,
+  main_email: string
+}
+
+const firestore = getFirestore();
+
+const findCustomerById = async (id: string | undefined) => {
+  if (id) {
+    try {
+      const customer = (await api.get(`customers/${id}`)).data;
+      if (customer) {
+        return customer as Customers;
+      }
+      return null;
+    } catch (e) {
+      logger.error(e);
+      return null;
+    }
+  }
+  return null;
+};
+
+export default async (customer: CustomerCheckout) => {
+  const returnCustomerAndAcessToken = async (custumer: Customers | null) => {
+    if (custumer) {
+      const authCustomer = await generateAccessToken(firestore, custumer._id);
+      return {
+        _id: custumer._id,
+        access_token: authCustomer.access_token,
+      };
+    }
+    return null;
```

Comment:
Pensando sobre isso aqui agora, 
Aqui vai é precisar criar um novo customer na Api e retornar com a autenticação dele né?

Reply:
Na verdade não, isso vai ser feito só na API do Passport mesmo

Reply:
"Não vai ter mais a possibilidade" de bater o checkout sem o cliente existir, o checkout deve exigir o token específico do cliente, pra ter um token pro cliente é necessário que o cliente exista (😛), quem cria o cliente (se ainda não houver pelo e-mail) e gera o token que depois vai ser enviado já pronto pro endpoint de checkout

Reply:
As aspas são porque o endpoint tá exposto e a request vem client side, mas o checkout deve usar a autenticação que vem no cabeçalho pra bater a Store API, se a autenticação não for válida (a.k.a. cliente não existe ou o token não foi gerado corretamente pelo Passport), já na primeira request pra Store API o checkout vai receber um erro (401) e deve repassar esse erro para o usuário, encerrando a requisição..